.github/workflows/build_mpich_source.yml

```diff
@@ -80,3 +83,13 @@ jobs:
             --enable-threads=multiple
           make -j2
           make install
+
+      - name: Tar MPICH installation to preserve permissions for artifact
+        run: tar -cvf mpich.tar ${{ runner.workspace }}/mpich
```

Comment:
Needed as a workaround. The default artifact zipping method doesn't preserve permissions, so GitHub mentions tar'ing first as a workaround.